apps/web/src/components/WalletConnect/WalletConnectProvider.tsx

```diff
@@ -0,0 +1,156 @@
+      handleAsyncActionUnsafe(async () => {
+        // dApp sends in the session proposal the required networks and the optional networks.
+        // The response must contain all the required networks but Umami supports just one per request.
+        // So if the list of required networks is more than one or the required network is not supported, we can only reject the proposal.
+        const requiredNetworks = Object.entries(proposal.params.requiredNamespaces)
+          .map(([key, values]) => (key.includes(":") ? key : (values.chains ?? [])))
+          .flat()
+          .filter(Boolean);
+
+        if (requiredNetworks.length !== 1) {
+          throw new Error(
+            `Umami supports only one network per request, got required networks: ${requiredNetworks}`
+          );
+        }
+        const network = requiredNetworks[0] as NetworkType;
+        const availablenetworks = availableNetworks.map(network => network.name);
+        // the network contains a namespace, e.g. tezos:mainnet
+        if (!availablenetworks.includes(network.split(":")[1])) {
+          throw new Error(
+            `The requested required network "${network}" is not supported. Available: ${availablenetworks}`
+          );
+        }
+
+        await openWith(<SessionProposalModal network={network} proposal={proposal} />, {});
+        console.log("Session proposal from dApp", proposal, walletKit.getActiveSessions());
+      }).catch(async () => {
+        // dApp is waiting so we need to notify it
+        await walletKit.rejectSession({
+          id: proposal.id,
+          reason: getSdkError("UNSUPPORTED_CHAINS"),
+        });
+      }),
+    [availableNetworks, openWith, handleAsyncActionUnsafe]
+  );
+
+  const onSessionDelete = useCallback(
+    (event: WalletKitTypes.SessionDelete) => {
+      // by that time the session is already deleted from WalletKit so we cannot find the dApp name
+      console.log("Session deleted by dApp", event);
+      toast({
+        description: "Session deleted by peer dApp",
+        status: "info",
+      });
+    },
+    [toast]
+  );
+
+  const onSessionRequest = useCallback(
+    async (event: WalletKitTypes.SessionRequest) => {
+      try {
+        const activeSessions: Record<string, SessionTypes.Struct> = walletKit.getActiveSessions();
+        if (!(event.topic in activeSessions)) {
+          console.error("WalletConnect session request failed. Session not found", event);
+          throw new Error("WalletConnect session request failed. Session not found");
+        }
+
+        const session = activeSessions[event.topic];
+
+        console.log("Session request, event, session", event, session);
+        console.log("Session request from dApp", session);
+        toast({
+          description: `Session request from dApp ${session.peer.metadata.name}`,
+          status: "info",
+        });
+        throw new Error("Not implemented");
+      } catch (error) {
```

Comment:
Yes. It will trigger request rejection. Later when request handling is implemented, the exception will still lead to reject sent to the dApp. This part is already in place.
If we don't send a reject, dApp will be hanging. The basic implementation waits for 5 minutes before giving up and doesn't allow to send another request. So it's easier to send a reject and unblock.